### PR TITLE
Improvements to ConfigMgr validation errors.

### PR DESCRIPTION
**Pull request by jordanfilteau1995**
better error messages for const, enum, type; only string and number supported for const and enum, should support int64.

Signed-off-by: Jordan Filteau <jfilteau@rocketsoftware.com>

### CODE REVIEW

**Comment by jordanfilteau1995** [comment]
Example...

Validity Exceptions(s) with object at
  Validity Exceptions(s) with object at /zowe
    Validity Exceptions(s) with object at /zowe/setup
      Validity Exceptions(s) with object at /zowe/setup/certificate
        no matching enum value at /zowe/setup/certificate/type; expecting one of values [PKCS12, JCERACFKS] of type 'string'

  Validity Exceptions(s) with object at /zOSMF
    type 'integer' not permitted at /zOSMF/host; expecting type 'string'

unequal constant value at /components/izp/security/surrogateUser/super; expecting value 'IZPSRGSP' of type 'string'

**Comment by jordanfilteau1995** [comment]
@1000TurquoisePogs Ready for review.

File: c/jsonschema.c [hunk 1]
```diff
@@ -123,6 +123,7 @@ typedef struct PatternProperty_tag {
 
 typedef struct JSValueSpec_tag {
   int      typeMask;
+  int      type;
```
Comment (1000TurquoisePogs) [inline c/jsonschema.c:126]: this needs to be at the end of the struct to not wreck any plugin that depends upon the old struct.

**Review by 1000TurquoisePogs — APPROVED** [review]
merging this and then fixing the struct in a second pr.

**Pull request merged**